src/pages/profile/components/profile-form/use-profile-form.ts

```diff
@@ -0,0 +1,194 @@
+import {
+	BaseSyntheticEvent,
+	ChangeEvent,
+	useCallback,
+	useMemo,
+	useState,
+} from "react";
+import { Control, FieldErrors } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import {
+	CountryData,
+	defaultCountries,
+	parseCountry,
+} from "react-international-phone";
+
+import { SelectChangeEvent } from "@mui/material";
+import { FetchBaseQueryError } from "@reduxjs/toolkit/query";
+
+import { useAppForm } from "~/libs/hooks/index.ts";
+import { User } from "~/libs/types/user.ts";
+import { type UserAndProfile } from "~/libs/types/user-profile.type.ts";
+import { setUser } from "~/redux/auth/auth-slice.ts";
+import { useAppDispatch, useAppSelector } from "~/redux/hooks.ts";
+import {
+	useUpdateUserAndProfileMutation,
+	useUploadMutation,
+} from "~/redux/user/user-api.ts";
+
+import { profileValidation } from "./profile-schema.ts";
+
+type UseProfileFormReturnType = {
+	control: Control<UserAndProfile>;
+	errors: FieldErrors<UserAndProfile>;
+	handleChangePhone: (phone: string) => void;
+	handleClothingSizeChange: (event: SelectChangeEvent<string>) => void;
+	handleFileChange: (e: ChangeEvent<HTMLInputElement>) => void;
+	handleFormSubmit: (event: BaseSyntheticEvent) => void;
+	handleJeansSizeChange: (event: SelectChangeEvent<string>) => void;
+	handleShoeSizeChange: (event: SelectChangeEvent<string>) => void;
+	phone: string;
+	phoneCountries: CountryData[];
+	selectedClothingSize: null | string;
+	selectedFile: File | null | string;
+	selectedJeansSize: null | string;
+	selectedShoeSize: null | string;
+	serverError: string;
+	user: User | null;
+};
+
+const zero = 0;
+const requiredCodes = ["ua", "ca"];
```

Comment:
I assumme it could be reused in some other places too, lets move it general constant file